gateway.go

```diff
@@ -298,13 +302,33 @@ func (gw *Gateway) toWildcardQName(qName, zone string) string {
 
 // Gets the set of addresses associated with the first set of index keys
 // that is in the indexer.
-func (gw *Gateway) getMatchingAddresses(indexKeySets [][]string) []netip.Addr {
+func (gw *Gateway) getMatchingAddresses(indexKeySets [][]string) map[string][]string {
 	// Iterate over supported resources and lookup DNS queries
 	// Stop once we've found at least one match
 	for _, indexKeys := range indexKeySets {
 		for _, resource := range gw.Resources {
 			addrs := resource.lookup(indexKeys)
-			if len(addrs) > 0 {
+
```

Comment:
Just no.

Reply:
Yes, the slices to check are addrs["A"], addrs["AAAA"] and addrs["TXT"] now. Because TXT records can't be put in the same slice as ip addresses.